src/Compilers/CSharp/Portable/Binder/ContextualAttributeBinder.cs

```diff
@@ -64,5 +66,10 @@ internal Symbol AttributeTarget
                 return _attributeTarget;
             }
         }
+
+        protected override CSDiagnosticInfo SingleTypeNameNotFound(string whereText, DiagnosticBag diagnostics, Location location, AssemblySymbol forwardedToAssembly)
+        {
+            return base.SingleTypeNameNotFound($"{whereText}' or '{whereText}Attribute", diagnostics, location, forwardedToAssembly);
```

Comment:
This would make the code more complicated and for every possible case, another error would have to be added.